src/superface-test.ts

```diff
@@ -169,11 +200,39 @@ export class SuperfaceTest {
     record: boolean,
     processRecordings: boolean,
     inputVariables?: InputVariables,
+    recordingVersion?: string,
     beforeRecordingLoad?: ProcessingFunction
   ): Promise<void> {
-    if (!this.recordingPath) {
-      throw new RecordingPathUndefinedError();
+    if (record) {
+      const enable_reqheaders_recording =
+        this.nockConfig?.enableReqheadersRecording ?? false;
+
+      recorder.rec({
+        dont_print: true,
+        output_objects: true,
+        use_separator: false,
+        enable_reqheaders_recording,
+      });
+
+      debug('Recording HTTP traffic started');
+    } else {
+      await this.loadRecordings(
+        processRecordings,
+        inputVariables,
+        recordingVersion,
+        beforeRecordingLoad
+      );
```

Comment:
thought: Naming is hard I know. And I know it is in comment, but still feels wrong to have function that should do traffic recording, but then it loads old recordings. I assume this was added later. So this would be good to fix eventually, and to help with readability of the code.

Reply:
I took this in consideration in refactor. It should be visible when it's recording and when not.